### PR TITLE
tests: remove temp loader directory before and after test

### PR DESCRIPTION
**Pull request by Loonride**
**What kind of change does this PR introduce?**

test fix

**Did you add tests for your changes?**

No

**If relevant, did you update the documentation?**

No

**Summary**

The temp directory should be removed before and after the test

Edit: I also moved `init` tests to the `test` directory and improved how they handle generated files

**Does this PR introduce a breaking change?**

No

**Other information**

### CODE REVIEW

**Comment by anshumanv** [comment]
We should do this for init tests as well, since they also generate some files.

File: package.json [hunk 1]
```diff
@@ -121,6 +121,7 @@
     "jest-watch-typeahead": "^0.4.2",
     "lerna": "^3.20.2",
     "lint-staged": "^10.0.8",
+    "mkdirp": "^1.0.4",
```
Comment (evilebottnawi) [inline package.json:124]: Why do not use https://nodejs.org/api/fs.html#fs_fs_mkdir_path_options_callback with `recursive`?

File: test/init/auto/init-auto.test.js [hunk 2]
```diff
@@ -1,22 +1,33 @@
-/* eslint-disable node/no-unpublished-require */
 'use strict';
 
-const firstPrompt = 'Will your application have multiple bundles?';
 const fs = require('fs');
+const rimraf = require('rimraf');
 const { join } = require('path');
-const { run } = require('../../../../../test/utils/test-utils');
+const { run } = require('../..//utils/test-utils');
```
Comment (evilebottnawi) [inline test/init/auto/init-auto.test.js:6]: Double `//` :smile:

File: test/loader/loader.test.js [hunk 3]
```diff
@@ -4,15 +4,25 @@
...
 jest.setTimeout(200000);
 
 describe('loader command', () => {
+    afterAll(() => {
+        rimraf.sync(loaderPath);
+    });
+
+    afterAll(() => {
+        rimraf.sync(loaderPath);
+    });
```
Comment (jamesgeorge007) [inline test/loader/loader.test.js:24]: Duplicate?

File: test/loader/loader.test.js [hunk 4]
```diff
@@ -4,15 +4,25 @@
...
 
 const ENTER = '\x0D';
 const loaderName = 'test-loader';
+const loaderPath = join(__dirname, loaderName);
 
 // Since scaffolding is time consuming
 jest.setTimeout(200000);
 
 describe('loader command', () => {
+    afterAll(() => {
```
Comment (jamesgeorge007) [inline test/loader/loader.test.js:18]:
```suggestion
    beforeAll(() => {
```

Comment (Loonride) [reply test/loader/loader.test.js]: Good catch, thanks!

**Comment by jamesgeorge007** [comment]
Thanks

**Pull request merged**